sdk/azidentity/default_azure_credential.go

```diff
@@ -51,33 +53,44 @@ func NewDefaultAzureCredential(options *DefaultAzureCredentialOptions) (*Default
 	if err == nil {
 		creds = append(creds, envCred)
 	} else {
-		errMsg += err.Error()
+		errorMessages = append(errorMessages, fmt.Sprintf("EnvironmentCredential: %s", err.Error()))
 	}
 
 	msiCred, err := NewManagedIdentityCredential(&ManagedIdentityCredentialOptions{ClientOptions: options.ClientOptions})
 	if err == nil {
 		creds = append(creds, msiCred)
 		msiCred.client.imdsTimeout = time.Second
 	} else {
-		errMsg += err.Error()
+		errorMessages = append(errorMessages, fmt.Sprintf("ManagedIdentityCredential: %s", err.Error()))
 	}
 
 	cliCred, err := NewAzureCLICredential(&AzureCLICredentialOptions{TenantID: options.TenantID})
 	if err == nil {
 		creds = append(creds, cliCred)
 	} else {
-		errMsg += err.Error()
+		errorMessages = append(errorMessages, fmt.Sprintf("AzureCLICredential: %s", err.Error()))
+	}
+
+	errMsg := "\n"
+	for _, msg := range errorMessages {
+		errMsg += fmt.Sprintf("\t%s\n", msg)
```

Comment:
You could include the tab in each message and use `strings.Join` here

Reply:
I don’t know, I like scoping the tab to where the formatting happens 🤔

Reply:
Why 😄? The messages are displayed only after joining, so nobody will see a lone, weirdly indented message. Consider also that calling `strings.Join` would be more concise and clearer (to my eyes anyway), and would have prevented the bug with the `len` check (and future such bugs when this code is refactored)

Reply:
I do like strings.Join. I’ll come back with an idea.

Reply:
What about this? https://github.com/Azure/azure-sdk-for-go/pull/16844/commits/ee848b15dcdb42e93d4bdd06e69c2796a4d3ed15